qutebrowser/mainwindow/mainwindow.py

```diff
@@ -344,6 +344,12 @@ def closeEvent(self, e):
             quit_texts.append("{} {} running.".format(
                 tab_count,
                 "download is" if tab_count == 1 else "downloads are"))
+        # Check if any tabs have filled forms and skip
+        if 'forms' in confirm_quit:
+            for tab in self._tabbed_browser.widgets():
+                if not self._tabbed_browser.check_forms_can_be_destroyed(tab):
+                    e.ignore()
+                return
```

Comment:
Returning here means only the first tab will be checked - if any other is modified, `e.ignore()` will not be called and qutebrowser will thus quit.

I first thought the solution would be to just outdent `return` by one, but we also want to continue with the other questions if the quit wasn't aborted yet. Maybe add this after the if?

```python
if not e.isAccepted():
    return
```